bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java

```diff
@@ -233,11 +233,16 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         TeslaChannelSelector selector = TeslaChannelSelector.getValueSelectorFromChannelID(channelID);
 
         if (command instanceof RefreshType) {
-            if (isAwake()) {
-                // Request the state of all known variables. This is sub-optimal, but the requests get scheduled and
-                // throttled so we are safe not to break the Tesla SLA
-                requestAllData();
+            if (!isAwake()) {
+                logger.debug("Waking vehicle to refresh all data");
+                wakeUp();
             }
+
+            setActive();
```

Comment:
Hm, the code says:
```
        // vehicle is inactive in case
        // - it does not charge
        // - it has not moved in the observation period
```
So I would claim that a REFRESH does not change anything about the "active" state, only about the "awake" state.

Reply:
I don't get that comment

Calling setActive() just resets the inactivity/location timestamp, so that we again begin to poll the vehicle (instead of not polling to let it fall asleep/go offline)

Reply:
The method sets `isInactive = false`, which is wrong, because according to aboves definition, the car is still inactive (not charging, not moved).
Furthermore `lastLocationChangeTimestamp` is set by this method, which is also wrong, because there was no location change.

> instead of not polling to let it fall asleep/go offline

We have just sent a wake up command. Won't the vehicle wake up through it?

Reply:
Let me, again, state the problem I'm trying to solve:

1. Your car is offline/asleep, sitting in your garage
2. You unlock it, it transitions to online, the binding eventually notices and begins to poll for 5 minutes: https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java#L100
3. You forgot something in your house and quickly go to grab it, you return after 6 minutes
4. Since MOVE_THRESHOLD_INTERVAL_MINUTES has passed, we do not poll the car for another 20 minutes: https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java#L99
5. In the meanwhile, you drive off and the binding misses updates on your position etc. for this period of time

Instead, what you can do (if this change gets accepted), is to set up a mechanism that makes your openhab send a REFRESH command to any channel when your phone is connected via bluetooth to your car. That way, as soon as you enter the car again after step 3, the binding "resets" the inactivity period, making it notice that the car is now moving.

If you have other plans on how this should work, please let me know. This was the way it worked in 2.4 where I added that functionality and you also reviewed it and thought it to be fine.

Reply:
Ok, thanks for re-stating your problem. I think some confusion came from the fact that what you NEED is actually a "set active" functionality, while what "force refresh" suggests is to do a one-time refresh (and imho it should be seen as a feature and not a bug if the car falls asleep afterwards quickly again).

I failed to see that in case of your issue [this code](https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java#L788-L7999) is not executed at all.

So let's keep the code as you did it here, sorry for the confusion.